assets/src/edit-story/components/panels/textStyle/font.js

```diff
@@ -47,16 +47,53 @@ function FontControls({ selectedElements, pushUpdate }) {
   const fontFamily = getCommonValue(selectedElements, 'fontFamily');
   const fontSize = getCommonValue(selectedElements, 'fontSize');
   const fontWeight = getCommonValue(selectedElements, 'fontWeight');
+  const timeout = useRef(null);
 
   const {
     state: { fonts },
-    actions: { getFontWeight, getFontFallback },
+    actions: { maybeEnqueueFontStyle, getFontWeight, getFontFallback },
   } = useFont();
   const fontWeights = useMemo(() => getFontWeight(fontFamily), [
     getFontWeight,
     fontFamily,
   ]);
 
+  const loadFont = useCallback(
+    (family, callback) => {
+      // 0. Check browser support to load fonts using JavaScript
+      if (document?.fonts) {
+        const font = `0 ${family}`;
+
+        // 1. Check if font family was not loaded yet
+        if (!document.fonts.check(font)) {
+          // 2. Try to inject <link /> resource of the font family
+          maybeEnqueueFontStyle(family);
+          // 3. Schedule a re-check since we need to wait the <link /> injection
+          timeout.current = setTimeout(async () => {
```

Comment:
Why do we need timeout when we have `fonts.load` promise?

Reply:
It will probably be changed as soon as I add more control on maybeEnqueueFontStyle success/fail events, but I already explained the process of this in https://github.com/google/web-stories-wp/pull/1275#discussion_r411707389